espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/InvocationOperation.java

```diff
@@ -37,17 +33,17 @@ public Object apply(final Object applyOn) throws Exception {
 
         final AtomicReference<Object> ref = new AtomicReference<Object>();
         final AtomicReference<Exception> refEx = new AtomicReference<Exception>();
-        final MethodWithArguments method = findCompatibleMethod(applyOn);
+        final Method method = findCompatibleMethod(applyOn);
         InstrumentationRegistry.getInstrumentation().runOnMainSync(new Runnable() {
             @Override
             public void run() {
                 Object result;
                 try {
-                    if (method.getMethod().getReturnType().equals(Void.TYPE)) {
-                        method.invoke(applyOn);
+                    if (method.getReturnType().equals(Void.TYPE)) {
+                        ReflectionUtils.invoke(method, applyOn, argumentValues.toArray());
```

Comment:
this conversation to array won't work. See https://stackoverflow.com/questions/5374311/convert-arrayliststring-to-string-array

That is why unit tests are needed

Reply:
I did not get you. This conversion gives  Object[] which is what ReflectionUtils.invoke expects. It  is invoking the right methods for me

Reply:
I mean it is mandatory to set the size and type of the resulting array in toArray() method, for example `toArray(new Object[0]) `

Reply:
did not get you why is it mandatory. By default `toArray()` is giving Object array of the desired size. Is there any particular advantage in specifying Type and length here?

Reply:
I think it makes sense to also return values as an array instead of a list. It would also resolve the other comment

Reply:
Makes sense, changed it.